### ISSUE TITLE
text/template: AddParseTree can't replace empty templates, whereas Parse can

### ISSUE THREAD

**Issue by chowey** [golang/go#10990]
Given that I have an empty "test" template:

```go
tpl, err := template.New("main").Parse(`
    {{template "test" .}}
    {{define "test"}}{{end}}
`)
```

this works:
```go
tpl, err = tpl.New("test").Parse(`This is a test`)
// no error
```

and this fails:
```go
test, err := template.New("test").Parse(`This is a test`)
tpl, err = tpl.AddParseTree("test", test.Tree)
// error
```

I think the AddParseTree [conditional](https://github.com/golang/go/blob/master/src/text/template/template.go#L113-115) should be changed to:
```go
    if t.common != nil
        if old := t.tmpl[name]; old != nil && !parse.IsEmptyTree(old.Root) {
            return nil, fmt.Errorf("template: redefinition of template %q", name)
        }
    }
```

This would make the behavior of AddParseTree consistent with Parse.

**Comment by chowey** [comment]
Is this now fixed with 6d9df14fec6361a022c9732da0ac87fda7f9963f?

**Comment by chowey** [comment]
Just tested - yes this is fixed now.
